code/modules/power/generators/turbine.dm

```diff
@@ -315,55 +308,45 @@
 		return
 	return ..()
 
-/obj/machinery/power/turbine/interact(mob/user)
-
-	if( !Adjacent(user)  || (stat & (NOPOWER|BROKEN)) && (!issilicon(user)) )
-		user.unset_machine(src)
-		user << browse(null, "window=turbine")
-		return
-
-	var/t = "<TT><B>Gas Turbine Generator</B><HR><PRE>"
-
-	t += "Generated power : [round(lastgen)] W<BR><BR>"
-
-	t += "Turbine: [round(compressor.rpm)] RPM<BR>"
-
-	t += "Starter: [ compressor.starter ? "<A href='?src=[UID()];str=1'>Off</A> <B>On</B>" : "<B>Off</B> <A href='?src=[UID()];str=1'>On</A>"]"
-
-	t += "</PRE><HR><A href='?src=[UID()];close=1'>Close</A>"
-
-	t += "</TT>"
-	var/datum/browser/popup = new(user, "turbine", name)
-	popup.set_content(t)
-	popup.open()
-
-	return
-
-/obj/machinery/power/turbine/Topic(href, href_list)
+/obj/machinery/power/turbine/attack_hand(mob/user)
+	. = ..()
+	ui_interact(user)
+
+/obj/machinery/power/turbine/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "TurbineComputer", name, 400, 300, master_ui, state)
+		ui.open()
+
+/obj/machinery/power/turbine/ui_data(mob/user)
+	var/list/data = list()
+	data["compressor"] = compressor ? TRUE : FALSE
+	data["compressor_broken"] = (!compressor || (compressor.stat & BROKEN)) ? TRUE : FALSE
```

Comment:
```suggestion
	data["compressor_broken"] = (!compressor || (compressor.stat & BROKEN))
```
ternary isn't needed here since you're doing a boolean operation